app/controllers/miq_ae_class_controller.rb

```diff
@@ -389,50 +389,28 @@ def replace_right_cell(replace_trees = [])
     render :js => presenter.to_html
   end
 
-  def get_combo_xml(fields)
+  def build_type_options(fields)
     aetypes = MiqAeField.available_aetypes
-    combo_xml = Array.new
     fields.each do |fld|
-      xml = REXML::Document.load("")
-      xml << REXML::XMLDecl.new(1.0, "UTF-8")
-      # Create root element
-      root = xml.add_element("complete")
-      if fld['aetype'].blank? && (session[:field_data].blank? || session[:field_data][:aetype].blank?)
+      if fld['aetype'].blank? && session.fetch_path(:field_data, :aetype).blank?
         fld['aetype'] = "attribute"
-      elsif !session[:field_data].blank? && !session[:field_data][:aetype].blank?
-        fld['aetype'] = session[:field_data][:aetype]
+      elsif !session.fetch_path(:field_data, :aetype).blank?
+        fld['aetype'] = session.fetch_path(:field_data, :aetype)
```

Comment:
in that case fld['aetype'] must be already set, by previous edit session or when that field was originally added.